akka-stream-tests/src/test/java/akka/stream/javadsl/SourceTest.java

```diff
@@ -1157,4 +1158,24 @@ public void mustBeAbleToConvertToJavaInJava() {
         akka.stream.scaladsl.Source.empty();
     Source<Integer, NotUsed> javaSource = scalaSource.asJava();
   }
+
+  @Test
+  public void mustProperlyIterate() throws Exception {
+    // #iterate-simple
+    final Creator<Iterator<Boolean>> input =
+        new Creator<Iterator<Boolean>>() {
+          @Override
+          public Iterator<Boolean> create() {
+            return Iterables.cycle(false, true).iterator();
+          }
+        };
+
+    final CompletableFuture<List<Boolean>> future =
+        Source.fromIterator(input).grouped(10).runWith(Sink.head(), system).toCompletableFuture();
+
+    assertArrayEquals(
+        new Boolean[] {false, true, false, true, false, true, false, true, false, true},
+        future.get(1, TimeUnit.SECONDS).toArray());
```

Comment:
I'd prefer that it is either a somewhat realistic use case example, or it is the minimum possible, also avoiding using a third party library, blocking and utilities only available in tests.

I'd rather suggest something like:
```
  Source.fromIterator(() -> Arrays.asList(1, 2 ,3).iterator()).runForeach(System.out::println, system);
  // prints
  // 1
  // 2 
  // 3
```

Reply:
Add a minimal possible example for docs
Also, keep the mustProperlyIterate test in SourceTest.java (this is an analog of test from SourceSpec.scala)